src/types/ServiceListComponent.ts

```diff
@@ -14,15 +13,23 @@ export interface ServiceList {
 
 export interface ServiceItem extends ServiceOverview {
   namespace: string;
+  healthPromise: Promise<Health>;
+  health?: Health;
 }
 
-export const overviewToItem = (overview: ServiceOverview, namespace: string): ServiceItem => {
-  return {
+export const overviewToItem = (
+  overview: ServiceOverview,
+  namespace: string,
+  healthPromise: Promise<Health>
+): ServiceItem => {
+  const item: ServiceItem = {
     name: overview.name,
-    health: overview.health,
     istioSidecar: overview.istioSidecar,
-    namespace: namespace
+    namespace: namespace,
+    healthPromise: healthPromise
   };
+  healthPromise.then(h => (item.health = h));
```

Comment:
Missing catch ? if this promise fails, where is the error msg handled ?

Reply:
No, this is just an intermediate operation on the promise, it's not the "terminal" operation. The promise is consumed at the component level, and that's where the error is caught.